packages/backend/src/services/DashboardService/DashboardService.ts

```diff
@@ -533,25 +597,24 @@ export class DashboardService {
             projectUuid,
             user.userUuid,
         );
+        const spaceAccess = await this.spaceModel.getSpaceAccess(
+            space.space_uuid,
+        );
 
         if (
             user.ability.cannot(
                 'update',
                 subject('Dashboard', {
                     organizationUuid: space.organization_uuid,
                     projectUuid,
+                    isPrivate: space.is_private,
+                    access: spaceAccess,
                 }),
             )
         ) {
             throw new ForbiddenError();
         }
```

Comment:
The original space check was wrong. We need 2 checks for each dashboard. 
Can the user move the dashboard from the existing space?
Can the user move the dashboard to the new space?

Reply:
quick question here, why we check the first accessible space here? seems redundant.
also i'v updated the logic to check access to each dashboard, feel free to change it back if i'm breaking any existing logic here.